providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/steps/ProviderSteps.java

```
@@ -125,28 +125,26 @@ public void setupProvider(String providerType) throws IOException, InterruptedEx
                 .statusCode(200);
 
         // giving flagd a little time to start
-        Thread.sleep(30);
+        Thread.sleep(100);
         FeatureProvider provider =
                 new FlagdProvider(state.builder.resolverType(State.resolverType).build());
 
         OpenFeatureAPI api = OpenFeatureAPI.getInstance();
-        String providerName = providerType + Math.random();
```

I think the providername adds a little bit more separation to the execution, so each e2e test has its own provider, without any sideeffects from previous tests, wdyt should we maybe keep this?

---

Since we recreate the api for every test, I don't think that the providers for each test can interfere with each other. If we happen to have more than one provider per api/ more than one api, this would be a whole other problem

---

We do not recreate the API for e2e tests. We still utilize the singleton as the end user would use it.

---

True, I misread that. But maybe we should?
I will reintroduce names for the provider

---

E2E tests should be closer to the consumer, simulating how they are used, we should not fall back to separate API instances, we might hide problems, which we initially planned to detect with e2e tests